ouroboros-consensus/src/Ouroboros/Consensus/Node/ProtocolInfo/Byron.hs

```diff
@@ -96,13 +137,19 @@ protocolInfoByron (NumCoreNodes numCoreNodes) nid params gc mLeader =
       , pInfoInitState  = ()
       }
   where
-    proofOfCredentials :: CoreNodeId -> PbftLeaderCredentials -> PBftIsLeader PBftCardanoCrypto
-    proofOfCredentials nid' (PbftLeaderCredentials sk dlg) =
+    delegation :: Cardano.Genesis.GenesisDelegation
+    delegation = Cardano.Genesis.configHeavyDelegation gc
+    proofOfCredentials :: Cardano.Genesis.GenesisDelegation -> PbftLeaderCredentials -> PBftIsLeader PBftCardanoCrypto
+    proofOfCredentials dlg cred@(PbftLeaderCredentials sk cert) =
```

Comment:
Once you fix the algorithm for `genesisKeyCoreNodeId` then you will not need this first arg at all.

Reply:
Done.